deploy/modules/gcp/tidb-cluster/data.tf

```diff
@@ -21,7 +21,7 @@ data "external" "monitor_port" {
 locals {
   # TODO Update related code when node locations is avaiable in attributes of cluster resource.
   cmd_get_cluster_locations = <<EOT
-gcloud --project ${var.gcp_project} container clusters list --filter='name=${var.gke_cluster_name}' --format='json[no-heading](locations)' --region ${var.gke_cluster_location} | jq '.[0] | .locations |= join(",")'
+gcloud --project ${var.gcp_project} container clusters list --filter='name=${var.gke_cluster_name}' --format='json[no-heading](locations)' --region ${var.gke_cluster_location} | jq '{"locations": (if (. | length) > 0 then .[0].locations | join(",") else "" end) }'
```

Comment:
Is the else case correct? It seems like there will be no zones, but there should be at least 1

Reply:
This is used to pass the refresh phase to generate the plan. In the apply phrase, the data source will be read again to get actual locations of the cluster.

it's ok to return empty locations when the cluster does not exist. This data source will be refresh after the cluster is created, see this log:

```
...
module.tidb-operator.google_container_cluster.cluster: Still creating... [8m0s elapsed]
module.tidb-operator.google_container_cluster.cluster: Still creating... [8m10s elapsed]
module.tidb-operator.google_container_cluster.cluster: Creation complete after 8m13s [id=sysbench]
module.default-tidb-cluster.data.external.cluster_locations: Refreshing state...
```

However, after reading [this doc](https://www.terraform.io/docs/configuration/data-sources.html#data-resource-behavior), I think it's better to put the cluster sources in  `query` argument to defer the reading of this data source until the cluster is created.

I'll update this PR.

Reply:
It seems there is no way to reference cluster resource in tidb-cluster because cluster resource is in another module (`tidb-operator`). IMO this is enough because the data resource will be refreshed after the cluster is created.